File: DFS/N-Queens.cpp
#include <iostream>
#include <vector>
#include <string>
#include <stdlib.h>

using std::string;
using std::vector;


vector<vector<string>> solveNQueens(int n) {
    vector<vector<string>> res;
    // vector<string> temp;
    vector<int> loc(n, -1);

}

void dfs(int start, int n, vector<vector<string>> & res,  vector<int> & loc) {
    if (start == n) {
        pa
    }
    else {
        for (int i = start; i < n; i++) {
            for (int j = 0; j < start, j++) {

            }
            if loc[i] == loc[j] || abs(loc[i] - loc[j]) == abs(i - j)
        }
    }
}

int main()
{
    int n = 4;
    std::cout << abs(4-5) << "\n";
    return 0;
}

File: Backtracking/NQueens.py
from typing import List

class Solution:
    def solveNQueens(self, n: int) -> List[List[str]]:
        _temp = []
        res = []

        def _not_valid(r, c):
            for i, j in _temp:
                if j == c or abs(i-r) == abs(j-c):
                    return True
            return False
            
        def backtrack(r, n):
            if r >= n:
                _s = [None for _ in range(n)]
                for i, j in _temp:
                    _s[i] = '.' * j + 'Q' + '.' * (n-1-j)
                res.append(_s)
                return
            for j in range(n):
                if _not_valid(r, j):
                    continue
                _temp.append((r, j))
                backtrack(r+1, n)
                _temp.pop()
                
        backtrack(0, n)
        return res


if __name__ == '__main__':
    solu = Solution()
    n = 4
    res = solu.solveNQueens(n)
    print(res)